ledger-api/grpc-definitions/com/daml/ledger/api/v1/version_service.proto

```diff
@@ -80,3 +81,8 @@ message DeduplicationPeriodSupport {
   OffsetSupport offset_support = 1;
   DurationSupport duration_support = 2;
 }
+enum ParticipantDeduplicationSupport {
+  PARTICIPANT_DEDUPLICATION_NOT_SUPPORTED = 0;
+  PARTICIPANT_DEDUPLICATION_PARALLEL_ONLY = 1;
+  PARTICIPANT_DEDUPLICATION_SUPPORTED = 2;
+}
```

Comment:
@nicu-da @SamirTalwar-DA @andreaslochbihler-da : a few important comments

1. Please add documentation to the individual fields that is sufficient for clients to understand what they mean, and for implementors to know the constraints on the implementation. The Ledger API documentation so far is a bit spotty, but I believe we can and should change that at least for all new additions.
2. What would we loose when moving this feature descriptor into the `ExperimentalFeatures`, and ask clients to not rely on it? I'm asking as AFAIR the distinctions here are for testing purposes only. Keeping such details out of the API surface used by production applications is very helpful to ensure we can continue evolving our code.

Reply:
@meiersi-da 

Good point with the documentation, will add it.

For me, it feels a bit awkward to add it to the `ExperimentalFeatures`. It is not really an experimental feature but it describes a behavior that I can't think of a reason to expose it to the users. Following this trail of thoughts, should we add a new field similar to `ExperimentalFeatures` but which can be called `InternalFeatures` and we document it as not to be used by clients? Or should this be exposed through a new, testing-only, API?

Reply:
Think of the "experimental" in `ExperimentalFeatures` as a message to the user that they should not rely on the feature descriptors in there. The comment on the field also states that they can used for internal testing purposes only, which seems to be the case for this kind of descriptor.

So, I'd suggest sticking your descriptor into `ExperimentalFeatures` and clarify the actual state of the feature itself in a comment, if that's required.

Reply:
In principle, an `InternalFeatures` would make sense, but I'm not sure it's ROI pays off in terms of engineering time. It will introduce extra churn, and also be problematic when iterating on a feature in multiple iterations where it's state switches from evolving to stabilized to evolving.

Reply:
I agree with sticking it in the `ExperimentalFeatures` field. I think the main issue I have here is that not some of the fields I'm adding shouldn't even be a feature descriptor, but the test should be ledger specific and not as part of the `ledger-api-test-tool`.

Reply:
Glad to hear wrt `ExperimentalFeatures`. Looking forward to the PR.

I agree that our testing story in the `ledger-api-test-tool` is sub-optimal. I'd expect that both kv-committer and kv-participant teams will each individually take steps to clean that up after removing kv-utils from the `daml` repo and replacing sandbox-classic with sandbox-on-x; e.g., adding integration tests for kv-util specific behavior alongside its implementation.

Until then, it looks like a decent strategy to me to at least remove the complexity of the different invocation of the `ledger-api-test-tool` by introducing `ExperimentalFeatures` descriptors to cover differences in ledgers and/or state of implementation evolution.